core/quorum_validation.go

```diff
@@ -167,3 +178,117 @@ func (c *Core) getUnpledgeId(wt string) string {
 	}
 	return b.GetUnpledgeId()
 }
+
+/*
+ * Function to check whether the TokenState is pinned or not
+ * Input tokenId, index, resultArray, waitgroup,quorumList
+ */
+func (c *Core) checkTokenState(tokenId, did string, index int, resultArray []TokenStateCheckResult, wg *sync.WaitGroup, quorumList []string) {
+	defer wg.Done()
+	var result TokenStateCheckResult
+	result.Token = tokenId
+	//get the latest blockId i.e. latest token state
+	b := c.w.GetLatestTokenBlock(tokenId, token.RBTTokenType)
+	if b == nil {
+		c.log.Error("Invalid token chain block")
+		result.Error = fmt.Errorf("Invalid token chain block")
+		result.Message = "Invalid token chain block"
+		resultArray[index] = result
+	} else {
+		blockId, err := b.GetBlockID(tokenId)
+		if err != nil {
+			c.log.Error("Error fetching block Id", err)
+			result.Error = err
+			result.Message = "Error fetching block Id"
+			resultArray[index] = result
+		}
+		//concat tokenId and BlockID
+		tokenIDTokenStateData := tokenId + blockId
+		tokenIDTokenStateBuffer := bytes.NewBuffer([]byte(tokenIDTokenStateData))
+
+		//add to ipfs get only the hash of the token+tokenstate
+		tokenIDTokenStateHash, err := c.ipfs.Add(tokenIDTokenStateBuffer, ipfsnode.Pin(false), ipfsnode.OnlyHash(true))
+		if err != nil {
```

Comment:
We can simplify the logic we use empty returns at each error after updating resultArray

Reply:
REsolved